tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -426,56 +572,303 @@ def get_fd_counts():
-                                                    target_topic, 0,
-                                                    self.preallocated_nodes)
+        seq_consumer = FranzGoVerifiableSeqConsumer(
+            self.test_context, self.redpanda, target_topic, 0,
+            [self.preallocated_nodes[2]])
         seq_consumer.start(clean=False)
         seq_consumer.shutdown()
         seq_consumer.wait()
         assert seq_consumer.consumer_status.invalid_reads == 0
-        assert seq_consumer.consumer_status.valid_reads == stress_msg_count + msg_count_per_topic
-
-        self.logger.info("Entering leader balancer stress test")
-
-        # Enable the leader balancer and check that the system remains stable
-        # under load.  We do not leave the leader balancer on for most of the test, because
-        # it makes reads _much_ slower, because the consumer keeps stalling and waiting for
-        # elections: at any moment in a 10k partition topic, it's highly likely at least
-        # one partition is offline for a leadership migration.
-        self.redpanda.set_cluster_config({'enable_leader_balancer': True},
-                                         expect_restart=False)
-        lb_stress_period = 120
-        lb_stress_produce_bytes = expect_bandwidth * lb_stress_period
-        lb_stress_message_count = int(lb_stress_produce_bytes /
-                                      stress_msg_size)
-        fast_producer = FranzGoVerifiableProducer(
-            self.test_context,
+        assert seq_consumer.consumer_status.valid_reads >= fast_producer.produce_status.acked + msg_count_per_topic
+
+        self.free_preallocated_nodes()
+
+    def _run_omb(self, scale: ScaleParameters):
+
+        # 1.25GB/s is a rather gentle rate on 12*i3en.xlarge 12 core nodes
+        # when using 1000 partitions per shard, use it as a baseline.
+        # (OMB is not good at stress testing, the worker nodes start throwing
+        # exceptions if a cluster isn't keeping up with the rate requested
+        # by the workload, so we are not aiming to aggressively saturate the system)
+        producer_bw = (
+            (len(self.redpanda.nodes) * scale.node_cpus) / 144.0) * 1.25E9
+
+        # It is necessary to scale the producer+consumer counts as well
+        # as the total throughput: otherwise there are too few messages
+        # per producer, and OMB reports this as a very high E2E latency.
+        # Roughly node_cpus is of the right order of magnitude.
+        producer_count = scale.node_cpus
+        consumer_count = scale.node_cpus
+
+        # Don't tweak this without also adjusting payload_file
+        message_size = 4096
+
+        producer_rate = producer_bw / message_size
+
+        # For really high partition counts, the throughput can't keep up
+        # with what the cluster did at more modest density, and this
+        # causes OMB to start failing internally with 500s when it backs up,
```

Comment:
Curious, is it OMB clients that can't keep up, or redpanda itself slows down enough that OMB can't get the rate it wants?

I guess this is a case of 1 client consuming/producing to many partitions, with few messages in each which is quite inefficient (unlike say the case of a client consuming/producing to 1 partition out of a large number which isn't necessarily very bad at all).

Reply:
There's some uncertainty there: I was hitting cases where OMB itself was failing (out of heap space etc), but unclear how much that was because of OMB being inefficient in its per-partition queues vs. it experiencing a backlog from redpanda not keeping up.